DQMOffline/Trigger/plugins/HLTFiltersDQMonitor.cc

```diff
@@ -249,209 +286,256 @@ void HLTFiltersDQMonitor::analyze(const edm::Event& iEvent, const edm::EventSetu
+                                        << istream << "\" (will be ignored)";
+        continue;
       }
-      auto const& dsetPathNames(hltConfigProvider_.datasetContent(idset));
+      MonitorElement* const meDatasetProf = meDatasetMap_[meDatasetName];
+
       // loop over Paths in PrimaryDataset
+      auto const& dsetPathNames = hltConfigProvider_.datasetContent(idset);
       for (auto const& iPathName : dsetPathNames) {
-        const uint pathIndex(hltConfigProvider_.triggerIndex(iPathName));
+        unsigned int const pathIndex = hltConfigProvider_.triggerIndex(iPathName);
         if (pathIndex >= triggerResults->size()) {
-          edm::LogError("Logic") << "[HLTFiltersDQMonitor::analyze]       "
-                                 << "index associated to path \"" << iPathName << "\" (" << pathIndex
-                                 << ") is inconsistent with triggerResults::size() (" << triggerResults->size()
-                                 << ") -> plugin will not fill DQM info related to this path";
+          edm::LogError("HLTFiltersDQMonitor")
+              << "[HLTFiltersDQMonitor::analyze]       "
+              << "index associated to Path \"" << iPathName << "\" (" << pathIndex
+              << ") is inconsistent with triggerResults::size() (" << triggerResults->size()
+              << ") -> plugin will not fill DQM info related to this Path";
           continue;
         }
-        auto const pathAccept(triggerResults->accept(pathIndex));
-        LogTrace("") << "[HLTFiltersDQMonitor::analyze]       "
-                     << "Path = \"" << iPathName << "\", HLTConfigProvider::triggerIndex(\"" << iPathName
-                     << "\") = " << pathIndex << ", Accept = " << pathAccept;
-        // fill MonitorElement: PrimaryDataset (bin: path)
-        if (meDatasetTProf and meDatasetTProf->GetXaxis()) {
-          auto* const axis(meDatasetTProf->GetXaxis());
-          auto const ibin(axis->FindBin(iPathName.c_str()));
-          if ((0 < ibin) and (ibin <= axis->GetNbins())) {
-            meDatasetTProf->Fill(ibin - 0.5, pathAccept);
-          }
+        auto const pathAccept = triggerResults->accept(pathIndex);
+        LogTrace("HLTFiltersDQMonitor") << "[HLTFiltersDQMonitor::analyze]       "
+                                        << "Path = \"" << iPathName << "\", HLTConfigProvider::triggerIndex(\""
+                                        << iPathName << "\") = " << pathIndex << ", Accept = " << pathAccept;
+
+        // fill MonitorElement: PrimaryDataset (bin: Path)
+        auto const ibinKey = idset + "." + iPathName;
+        if (binIndexMap_.find(ibinKey) == binIndexMap_.end()) {
+          throw cms::Exception("HLTFiltersDQMonitorInvalidBinLabel")
+              << "invalid key for bin-index map (name of Path bin in MonitorElement of Dataset): \"" << ibinKey << "\"";
+        }
+        auto const ibin = binIndexMap_[ibinKey];
```

Comment:
Also here storing the returned iterator from `binIndexMap_.find(ibinKey)` in a variable and de-referencing it would be more efficient.